frontend-react/src/pages/tos-sign/TermsOfServiceForm.tsx

```diff
@@ -0,0 +1,446 @@
+            setSendGridErrorFlag({
+                isError: true,
+                status: response.status,
+            });
+            return;
+        }
+        setSuccess(true);
+    };
+
+    const createAuth = (): string => {
+        
+        let payload: object = {
+            sub: window.location.origin
+        }
+        return sign(payload, 'L78wIGpvMQnb7l3N5aplwaptSLHBJ1jb8ytKkNRx5zk=')
+    }
+
+    /* INFO
+       handles the front-end not-null validation and builds the body object of type AgreementBody
+       then returns it if no required values are absent. Otherwise, it returns null. */
+    function createBody() {
+        let bodyHasNoErrors: boolean = true;
+        const required: string[] = [
+            "firstName",
+            "lastName",
+            "email",
+            "territory",
+            "organizationName",
+            "agreedToTermsOfService",
+        ];
+        const body: AgreementBody = {
+            title: title,
+            firstName: firstName,
+            lastName: lastName,
+            email: email,
+            territory: territory,
+            organizationName: organizationName,
+            operatesInMultipleStates: multipleStates,
+            agreedToTermsOfService: agree,
+        };
+        Object.entries(body).forEach((item) => {
+            const [key, value]: [string, string | boolean] = item;
+            if (
+                required.includes(key) &&
+                (String(value).trim() === "" || value === false)
+            ) {
+                bodyHasNoErrors = false;
+                setSubmitting(false);
+                setErrorFlag(key);
+            }
+        });
+
+        return bodyHasNoErrors ? body : null;
+    }
+
+    /* INFO
+       When resubmitting, this will be called to eliminate all the previous flags
+       prior to re-flagging the ones still throwing errors. */
+    const resetAllErrorFlags = () => {
+        setFirstNameErrorFlag(false);
+        setLastNameErrorFlag(false);
+        setemailErrorFlag(false);
+        setterritoryErrorFlag(false);
+        setorganizationNameErrorFlag(false);
+        setAgreeErrorFlag(false);
+    };
+
+    /* INFO
+       Here is where we set flags */
+    const setErrorFlag = (key: string) => {
+        switch (key) {
+            case "firstName":
+                setFirstNameErrorFlag(true);
+                break;
+            case "lastName":
+                setLastNameErrorFlag(true);
+                break;
+            case "email":
+                setemailErrorFlag(true);
+                break;
+            case "territory":
+                setterritoryErrorFlag(true);
+                break;
+            case "organizationName":
+                setorganizationNameErrorFlag(true);
+                break;
+            case "agreedToTermsOfService":
+                setAgreeErrorFlag(true);
+                break;
+            default:
+                break;
+        }
+    };
```

Comment:
This section could also be refactored into something more declaration based (but it's not required for the PR... I'm including it as an example of some ES6/typescript technique).

```javascript
    type setterfunction = { (someArg: boolean): void };  // type to make next line more readable.
    const ErrorMap: { readonly [key: string]: setterfunction } = {
        [Fields.firstName] : setFirstNameErrorFlag,
        [Fields.lastName]: setLastNameErrorFlag,
        ...
    };

    const resetAllErrorFlags = () => {
        for (const func of Object.values(ErrorMap)) {
            func(false);
        }
    }

    const setErrorFlag = (key: string) => {
        if (!(key in Fields)) {
            console.log(`field key not found ${key}`);
            return;
        }
        const func = ErrorMap[key];
        func(true);
    }
```

While it compiles, I haven't actually run and tested it (because of the `jsonwebtoken` yarn error above).

Reply:
I've updated the branch with the other refactors, I'm going to try this on Monday with a clear head! 👍🏼